lib/ec2/EC2.rb

```diff
@@ -10,28 +10,230 @@ class << self
-      # Public:
+      require "aws_extensions/ec2/Vpc"
+      Aws::EC2::Types::Vpc.send(:include, AwsExtensions::EC2::Vpc)
+
+      require "aws_extensions/ec2/RouteTable"
+      Aws::EC2::Types::RouteTable.send(:include, AwsExtensions::EC2::RouteTable)
+
+      require "aws_extensions/ec2/NetworkAcl"
+      Aws::EC2::Types::NetworkAcl.send(:include, AwsExtensions::EC2::NetworkAcl)
+
+      require "aws_extensions/ec2/NetworkInterface"
+      Aws::EC2::Types::NetworkInterface.send(:include, AwsExtensions::EC2::NetworkInterface)
+
+      # Public
       #
-      # Returns a map of subnets mapped to their id
+      # Returns a Hash of subnets mapped to their id
       def id_subnets
         @id_subnets ||= Hash[subnets.map { |subnet| [subnet.subnet_id, subnet] }]
       end
 
-      # Public:
+      # Public
       #
-      # Returns a map of subnets mapped to the value of the "Name" tag
+      # Returns a Hash of subnets mapped to the value of the "Name" tag
       def named_subnets
         @named_subnets ||= Hash[subnets.map { |subnet| [subnet.name, subnet] }]
           .reject { |k, v| k.nil? or v.nil? }
       end
 
-      # Public:
+      # Public
       #
-      # Returns all subnets as an array of Aws::EC2::Types::Subnet
+      # Returns a Hash of subnets mapped to their vpc id
+      def vpc_subnets
+        @vpc_subnets ||= Hash[id_vpcs.map do |vpc_id, _|
+          [vpc_id, subnets.select { |subnet| subnet.vpc_id == vpc_id }]
+        end]
+      end
+
+      # Public: Lazily load the subnets
       def subnets
         @subnets ||= init_subnets
       end
 
+      # Public
+      #
+      # Returns a Hash of Aws::EC2::Types::Vpc to a VPC's name or id if there is no name
+      def named_vpcs
+        @named_vpcs ||= Hash[vpcs.map { |vpc| [vpc.name || vpc.vpc_id, vpc] }]
+          .reject { |k, v| k.nil? or v.nil? }
+      end
+
+      # Public
+      #
+      # Returns a Hash of Aws::EC2::Types::Vpc to a VPC's id
+      def id_vpcs
+        @vpc_ids ||= Hash[vpcs.map { |vpc| [vpc.vpc_id, vpc] }]
+      end
+
+      # Public refreshes the list of vpcs
+      def refresh_vpcs!
+        @vpcs = init_vpcs
+      end
+
+      # Public: Lazily load the vpcs
+      def vpcs
+        @vpcs ||= init_vpcs
+      end
+
+      # Public
+      #
+      # Returns a Hash of Aws::EC2::Types::RouteTable to that route tables name or id if there is no name
+      def named_route_tables
+        @named_route_tables ||= Hash[route_tables.map { |rt| [rt.name || rt.route_table_id, rt] }]
+          .reject { |k, v| k.nil? or v.nil? }
+      end
+
+      # Public
+      #
+      # Returns a Hash of Aws::EC2::Types::RouteTable to each subnet id associated with the route table
+      def subnet_route_tables
+        @subnet_route_tables ||= Hash[route_tables.flat_map do |rt|
+          rt.subnet_ids.map { |subnet_id| [subnet_id, rt] }
+        end]
+      end
+
+      # Public
+      #
+      # Returns a Hash of Aws::EC2::Types::RouteTable arrays to vpc id
```

Comment:
I believe the relationship is the opposite, ie, vpc id to array of corresponding RouteTables